File: Code 3 Callouts/Code 3 Callouts/Models/Callouts/ICalloutBase.vb
Imports Rage
Imports Stealth.Plugins.Code3Callouts.Models.Peds
Imports Stealth.Plugins.Code3Callouts.Util.Audio

Namespace Models.Callouts

    Public Interface ICalloutBase

        Sub DisplayObjective()
        Function CalculateSpawnpoint() As Vector3
        Function PerformPedChecks() As Boolean
        Sub OnArrivalAtScene()
        Sub CreateBlip()
        Sub DeleteBlip()
        Sub OfficerDown()
        Sub DeleteEntities()
        Sub AskPedToFollowOfficer()

        Property CalloutID As Guid
        Property Objective As String
        Property CallDispatchTime As DateTime
        Property ResponseType As CallResponseType
        ReadOnly Property IsFixedSpawnPoint As Boolean
        ReadOnly Property RequiresSafePedPoint As Boolean
        Overloads Property State As CalloutState
        Property CallBlip As Blip
        Property CallDetails As String
        Property Markers As List(Of Blip)
        ReadOnly Property ShowAreaBlipBeforeAccepting As Boolean
        Property SkipRespondingState() As Boolean

        Property PedsToIgnore As List(Of Rage.PoolHandle)

        Property FoundPedSafeSpawn As Boolean

    End Interface

End Namespace

File: Code 3 Callouts/Code 3 Callouts/Models/Callouts/ICalloutBase.cs
using Rage;
using Stealth.Plugins.Code3Callouts.Models.Peds;
using Stealth.Plugins.Code3Callouts.Util.Audio;
using System;
using System.Collections.Generic;
using static Stealth.Plugins.Code3Callouts.Common;

namespace Stealth.Plugins.Code3Callouts.Models.Callouts
{

	internal interface ICalloutBase
	{

		void DisplayObjective();
		Vector3 CalculateSpawnpoint();
		bool PerformPedChecks();
		void OnArrivalAtScene();
		void CreateBlip();
		void DeleteBlip();
		void OfficerDown();
		void DeleteEntities();

		void AskPedToFollowOfficer();
		Guid CalloutID { get; set; }
		string Objective { get; set; }
		DateTime CallDispatchTime { get; set; }
		CallResponseType ResponseType { get; set; }
		bool IsFixedSpawnPoint { get; }
		bool RequiresSafePedPoint { get; }
		CalloutState CalloutState { get; set; }
		Blip CallBlip { get; set; }
		string CallDetails { get; set; }
		List<Blip> Markers { get; set; }
		bool ShowAreaBlipBeforeAccepting { get; }

		bool SkipRespondingState { get; set; }

		List<Rage.PoolHandle> PedsToIgnore { get; set; }

		bool FoundPedSafeSpawn { get; set; }
	}

}